src/main/thrift/parquet.thrift

```diff
@@ -467,6 +467,35 @@ struct SchemaElement {
   10: optional LogicalType logicalType
 }
 
+struct SchemaElementV3 {
```

Comment:
Antoine, on the question of keeping implementation simpler, would it pay to not revise this?

Reply:
We can, of course, but deprecated fields will probably never be removed from the format (except perhaps in 20 years).

Reply:
Are deprecated fields a problem?

Assuming a writer writes them, the writer wastes cpu and bytes.
A reader can chooce to parse them or ignore them (by removing/commenting them out) from the `.thrift` file. The latter means the deprecated fields will be ignored by the thrift parser.

Reply:
There is no mandate. The official parquet thrift will comment them out.
1. Writers compiled with old version of the official thrift file may write the fields.
2. Writers compiled with new version of the official thrift file won't write the fields.
3. Readers compiled with old version of the official thrift file may read the fields.
4. Readers compiled with new version of the official thrift file will ignore the fields.
5.

Reply:
The problem is that, for any given implementation, we'll have either 1+3 (compiled with old version: backwards compatibility but no perf improvement when reading), or 2+4 (compiled with new version: better performance on read, but backwards compatibility is lost).

This doesn't satisfy the requirements we're trying to satisfy here.

Reply:
Oh I see. The problem is old readers and new writers.

Reply:
I'm OK with no cleanup of deprecated fields.  Marking them semantically as logically required for `PAR1` footers seems reasonable to me.  I hope at some point standalone PAR3 would become the default and they would not need to be written if they present meaningful overhead once that happens.

from a wire compatibility perspective required->optional should be forward/backward compatible.